From 7f67a906f2d1d385d8cc97e37d5967d384e01bc4 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Sat, 15 May 2021 20:55:39 +0200
Subject: [PATCH 1/1] fs: fat: allow retrieving a dent for a directory

When resolving a directory the iterator is always descending into
the directory. For retrieving directory attributes it is necessary to keep
the pointer to the directory entry instead of setting it to NULL.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 fs/fat/fat.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/fs/fat/fat.c b/fs/fat/fat.c
index c561d82b35..0b37d7d101 100644
--- a/fs/fat/fat.c
+++ b/fs/fat/fat.c
@@ -793,7 +793,6 @@ static void fat_itr_child(fat_itr *itr, fat_itr *parent)
 		itr->start_clust = parent->fsdata->root_cluster;
 		itr->is_root = 1;
 	}
-	itr->dent = NULL;
 	itr->remaining = 0;
 	itr->last_cluster = 0;
 }
-- 
2.30.2

